versions/3.1.1.md

```diff
@@ -2722,8 +2726,7 @@ MyResponseType:
   - $ref: '#/components/schemas/Lizard'
 ```
 
-which means the payload _MUST_, by validation, match exactly one of the schemas described by `Cat`, `Dog`, or `Lizard`.  In this case, a discriminator MAY act as a "hint" to shortcut validation and selection of the matching schema which may be a costly operation, depending on the complexity of the schema. We can then describe exactly which field tells us which schema to use:
-
+which means the payload _MUST_, by validation, match exactly one of the schemas described by `Cat`, `Dog`, or `Lizard`. Evaluating a `oneOf` can be a costly operation, so `discriminator` MAY be used as a "hint" to improve the efficiency of selection of the matching schema. The `discriminator` keyword cannot change the validation result of the `oneOf`, it can only help make the evaluation more efficient and provide better error messaging. We can then describe exactly which field tells us which schema is expected to match the instance:
```

Comment:
The description of the PR:
> The discriminator keyword can't be a hint and a shortcut.

but here in the spec I see:
> `discriminator` MAY be used as a "hint" to improve the efficiency of selection of the matching schema

Probably it's more about the PR description, but I think it still worth pointing out 😅

Reply:
I'm not following what you're pointing out. My goal with this change is to make it clear that `discriminator` doesn't change the validation behavior of the `oneOf`/`anyOf` it applies to. At best, it can be used internally by the validator to improve validation performance. For example, if you have a discriminated `anyOf` and the schema you expect to pass is the fifth schema listed, then `discriminator` can be used to tell the implementation to check the fifth schema first deferring the first, second, third, and fourth schemas that aren't expected to pass. But, if that fifth schema fails validation, the validator still needs to check the rest of the schemas to look for a match. So, the validation isn't shortcutted. You still have to do the same validation checks as you would if `discriminator` wasn't present. `discriminator` allows the validator to _prioritize_ the evaluation of a schema that is expected to pass, it doesn't _eliminate_ the other schemas from consideration.

Reply:
Yup-yup, I see, and since `discriminator` MAY be used as a hint, description of the PR should be changed from:
> The discriminator keyword can't be a hint and a shortcut.
to maybe smth like this:
> The discriminator keyword can't be a shortcut, only a hint.

Reply:
```suggestion
which means the payload _MUST_, by validation, match exactly one of the schemas described by `Cat`, `Dog`, or `Lizard`. Evaluating a `oneOf` can be a costly operation, so `discriminator` MAY be used as a "hint" to improve the efficiency of selection of the matching schema. The `discriminator` keyword cannot change the validation result of the `oneOf`, it can only help make the evaluation more efficient and provide better error messaging. We can specify the exact field that tells us which schema is expected to match the instance:
```